base/uk.ac.stfc.isis.ibex.e4.product/src/uk/ac/stfc/isis/ibex/e4/product/ApplicationWorkbenchWindowAdvisor.java

```diff
@@ -108,6 +109,25 @@ public void postWindowCreate() {
         performMultipleClientsCheck();
     }
     
+    /**
+     * Remove the current PID from the file.
+     */
+    @Override
+    public void postWindowClose() {
```

Comment:
The ticket says "An alternative should be used that avoid needing to be explicitly cleaned up", but this is explicit clean-up code that might not run (e.g. if machine is powered off suddenly, client is killed via task manager, client crashes with a system exception).

Could you investigate using the `FileLock` api as a replacement for this approach, as I think that will be much more robust against all of the above failure conditions.